shoop/admin/modules/products/views/forms.py

```diff
@@ -186,7 +186,15 @@ def __init__(self, **kwargs):
             self.fields["shops"].initial = [default_shop]
 
         self.file_url = self.instance.url
-        self.thumbnail = None
+
+        if self.instance and self.instance.file:
+            try:
+                thumbnail = self.instance.get_thumbnail()
+                self.thumbnail = thumbnail.url
+            except Exception:
+                self.thumbnail = None  # no thumbnail available for whatever reason
+        else:
+            self.thumbnail = None
```

Comment:
1. All this iffing and try-execpting should be in the ProductMedia.get_thumbnail, since it is ProductMedia's own business to know when it can't be thumbnailed or how the thumbnailing can fail (and return None in that case).
2. The catch is still too broad. Find out the exact exceptions easy_thumbnails_thumbnailer.get_thumbnail will throw when it fails and catch only those, not every Exception.
3. You're using self.instance.url above so the `if sefl.instance` here is probably not needed (or it should be done in the `self.file_url` too?).
Also,